generate-stackbrew-library.sh

```diff
@@ -57,52 +62,63 @@ join() {
 	echo "${out#$sep}"
 }
 
-for variant in debian alpine; do
-	commit="$(dirCommit "$variant")"
-
-	fullVersion="$(git show "$commit":"$variant/Dockerfile" | awk '$1 == "ENV" && $2 == "MEMCACHED_VERSION" { print $3; exit }')"
-
-	versionAliases=()
-	while [ "${fullVersion%.*}" != "$fullVersion" ]; do
-		versionAliases+=( $fullVersion )
-		fullVersion="${fullVersion%.*}"
+for version; do
+	export version
+
+	variants="$(jq -r '.[env.version].variants | map(@sh) | join(" ")' versions.json)"
+	eval "variants=( $variants )"
+
+	alpine="$(jq -r '.[env.version].alpine' versions.json)"
+	debian="$(jq -r '.[env.version].debian' versions.json)"
```

Comment:
I don't see these two used anywhere. I think they are supposed to be for choosing the default debian/alpine tag per version?  🤔 oh, maybe line 90 and 101 are supposed to be these variables instead of a string?

I ran `./generate-stackbrew-library.sh` locally and it looks a bit odd. I usually do a double check by running `diff -u ./path/to/library/memcached <(./generate-stackbrew-library.sh)`

```diff
--- ../official-images/library/memcached      2023-08-04 12:13:58.543257371 -0700
+++ /dev/fd/63  2023-10-17 17:07:32.174814309 -0700
@@ -1,15 +1,15 @@
-# this file is generated via https://github.com/docker-library/memcached/blob/3e552dc7c3907467c258c986cb4763a544ee3041/generate-stackbrew-library.sh
+# this file is generated via https://github.com/docker-library/memcached/blob/da70bf65ad1bf6151dd1efe0d4c54d4766730b5b/generate-stackbrew-library.sh

 Maintainers: Tianon Gravi <admwiggin@gmail.com> (@tianon),
              Joseph Ferguson <yosifkit@gmail.com> (@yosifkit)
 GitRepo: https://github.com/docker-library/memcached.git

-Tags: 1.6.21, 1.6, 1, latest, 1.6.21-bookworm, 1.6-bookworm, 1-bookworm, bookworm
+Tags: 1.6.22-bookworm, 1.6-bookworm, 1-bookworm, bookworm, 1.6.22-bookworm-bookworm, 1.6-bookworm-bookworm, 1-bookworm-bookworm, bookworm-bookworm
 Architectures: amd64, arm32v5, arm32v7, arm64v8, i386, mips64le, ppc64le, s390x
-GitCommit: d3a51632e123ddcbf1b1e06e6cefdaa65f27b29a
-Directory: debian
+GitCommit: da70bf65ad1bf6151dd1efe0d4c54d4766730b5b
+Directory: 1.6/bookworm

-Tags: 1.6.21-alpine, 1.6-alpine, 1-alpine, alpine, 1.6.21-alpine3.18, 1.6-alpine3.18, 1-alpine3.18, alpine3.18
+Tags: 1.6.22-alpine3.18, 1.6-alpine3.18, 1-alpine3.18, alpine3.18, 1.6.22-alpine3.18-3.18, 1.6-alpine3.18-3.18, 1-alpine3.18-3.18, alpine3.18-3.18
 Architectures: amd64, arm32v7, arm64v8, i386, ppc64le, s390x
-GitCommit: d3a51632e123ddcbf1b1e06e6cefdaa65f27b29a
-Directory: alpine
+GitCommit: da70bf65ad1bf6151dd1efe0d4c54d4766730b5b
+Directory: 1.6/alpine3.18
```